api/saml/configuration.py

```diff
@@ -0,0 +1,376 @@
+import pickle
+
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
+
+from api.saml.exceptions import SAMLError
+from api.saml.metadata import ServiceProviderMetadata, IdentityProviderMetadata
+from core.model import ConfigurationSetting
+from flask_babel import lazy_gettext as _
+
+
+class SAMLConfigurationSerializingError(SAMLError):
+    """Raised in the case of any errors during configuration serializing"""
+
+
+class SAMLConfigurationSerializer(object):
+    """Serializes and deserializes values as library's configuration settings"""
+
+    def __init__(self, integration):
+        """Initializes a new instance of SAMLConfigurationSerializer class
+
+        :param integration: External integration
+        :type integration: ExternalIntegration
+        """
+        self._integration = integration
+
+    def serialize(self, setting_name, value):
+        """Serializes the value as a pickled library's configuration setting
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :param value: Value to be serialized
+        :type value: Any
+        """
+        ConfigurationSetting.for_externalintegration(
+            setting_name,
+            self._integration).value = value
+
+    def deserialize(self, setting_name):
+        """Deserializes and returns the library's configuration setting
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :return: Any
+        """
+        value = ConfigurationSetting.for_externalintegration(
+            setting_name,
+            self._integration).value
+
+        return value
```

Comment:
Reading this I was a little confused by your use of "serialize" to describe the act of taking a value of a configuration setting out of the database _and converting it from a string to a usable value_. To the extent we talk about "serializing" we generally use it to refer to the second part. I don't think this is a big deal, but it led to some odd moments for me later on.